### PR TITLE
[WIP] Add deskewing per tesseract

### PR DESCRIPTION
**Pull request by wrznr**
With the help of `OSD`, tesseract determines the skew angle for
images. The wrapper applies this to pages and regions. It is
not clear yet how to save the estimated skew angle in PAGE
XML. Cf. https://github.com/PRImA-Research-Lab/PAGE-XML/issues/9

### CODE REVIEW

File: ocrd_tesserocr/deskew.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        # print(self.parameter)
+        oplevel = self.parameter['operation_level']
+        with PyTessBaseAPI(path=TESSDATA_PREFIX, psm=PSM.AUTO_OSD) as tessapi:
+            for (n, input_file) in enumerate(self.input_files):
+                log.info("INPUT FILE %i / %s", n, input_file)
+                pcgts = page_from_file(self.workspace.download_file(input_file))
+                pil_image = self.workspace.resolve_image_as_pil(pcgts.get_Page().imageFilename)
+
+                metadata = pcgts.get_Metadata() # ensured by from_file()
+                metadata.add_MetadataItem(
```
Comment (wrznr) [inline ocrd_tesserocr/deskew.py]: I copied the `metadata` code from https://github.com/OCR-D/ocrd_tesserocr/blob/d968abcd62dab1f862faa7df4e1676d70322402f/ocrd_tesserocr/recognize.py#L106 ff. No idea what it actually does or if it is necessary. In addition, `segment_region` does not have it.

**Comment by wrznr** [comment]
@bertsky This means: If I iterate over the blocks and determine their skew, the first block will yield an identical value? I will check this. If it tuns out to be the case, I will restrict the operation on blocks, okay?

**Comment by bertsky** [comment]
@wrznr Yes exactly, that what I would expect. Yes, `RIL.BLOCK` should be good (via simple for loop on `iterate_level` or using `Next` and `IsAtFinalElement`, as in recognize.py).

**Comment by wrznr** [comment]
`_process_regions` is already there!

**Comment by bertsky** [comment]
> `_process_regions` is already there!

Right, this may work, too. (I still had on my mind what the API suggests: using the blocks from its PageIterator, not from an external layout analysis. We had a similar situation with text recognition BTW: results may be worse if the engine does not get to see the kinds of layout it was trained on. Maybe one should try to compare both modes empirically, just to be sure how robust this is?)

**Comment by wrznr** [comment]
@bertsky Is it necessary to update this PR or do your forthcoming additions to ocrd_tesserocr contain this proposed functionalities?

**Comment by bertsky** [comment]
@wrznr I started out independently, but I could of course rebase on this PR and either add or squash and force-push. It keeps making more sense by the minute :-)

**Comment by bertsky** [comment]
Sorry, I rebased this on the current master, but was not subsequently allowed to force-push here. So I started a new PR instead (#48), which supersedes this one.

**Pull request closed without merging**